Code/Framework/AzFramework/AzFramework/Quality/QualitySystemComponent.cpp

```diff
@@ -46,37 +51,37 @@ namespace AzFramework
 
     void QualitySystemComponent::Activate()
     {
-        m_settingsRegistry = AZ::SettingsRegistry::Get();
-        AZ_Assert(m_settingsRegistry, "QualitySystemComponent requires a SettingsRegistry but no instance has been created.");
+        auto settingsRegistry = AZ::SettingsRegistry::Get();
+        AZ_Assert(settingsRegistry, "QualitySystemComponent requires a SettingsRegistry but no instance has been created.");
 
-        m_console = AZ::Interface<AZ::IConsole>::Get();
-        AZ_Assert(m_console, "QualitySystemComponent requires an IConsole interface but no instance has been created.");
+        auto console = AZ::Interface<AZ::IConsole>::Get();
+        AZ_Assert(console, "QualitySystemComponent requires an IConsole interface but no instance has been created.");
 
-        if (m_settingsRegistry && m_console)
+        if (settingsRegistry && console)
         {
-            QualitySystemEvents::Bus::Handler::BusConnect();
-            m_settingsRegistry->Get(m_defaultGroupName, QualitySettingsDefaultGroupKey);
+            settingsRegistry->Get(m_defaultGroupName, QualitySettingsDefaultGroupKey);
             RegisterCvars();
+            QualitySystemEvents::Bus::Handler::BusConnect();
         }
     }
 
     void QualitySystemComponent::Deactivate()
     {
         QualitySystemEvents::Bus::Handler::BusDisconnect();
         m_settingsGroupCVars.clear();
-        m_qualityGroupStack.clear();
     }
 
     void QualitySystemComponent::LoadDefaultQualityGroup(int qualityLevel)
     {
         if (m_defaultGroupName.empty())
         {
             AZ_Warning("QualitySystemComponent", false, "No default quality group settings will be loaded because no default group name was defined at %s", QualitySettingsDefaultGroupKey);
-            return;
         }
-
-        m_console->PerformCommand(AZStd::string_view(m_defaultGroupName),
-            { AZStd::to_string(qualityLevel).c_str() }, AZ::ConsoleSilentMode::Silent);
+        else if (auto console = AZ::Interface<AZ::IConsole>::Get(); console != nullptr)
+        {
+            console->PerformCommand(AZStd::string_view(m_defaultGroupName),
+                { AZStd::to_string(qualityLevel).c_str() }, AZ::ConsoleSilentMode::Silent);

```

Comment:
Well there is no need to go to a string here, since the PerformCommand function accepts a string view for the arguments.
Also this can use `AZ::ConsoleTypeHelpers::ToString` as well

```c++
 console->PerformCommand(m_defaultGroupName,
                { AZ::ConsoleTypeHelpers::ToString(qualityLevel) }, AZ::ConsoleSilentMode::Silent);
```

Reply:
Good catch, I need to switch that function to use the enum type anyway